apps/api/v2/src/ee/me/services/me.service.ts

```diff
@@ -0,0 +1,74 @@
+import { SchedulesService_2024_04_15 } from "@/ee/schedules/schedules_2024_04_15/services/schedules.service";
+import { PrismaFeaturesRepository } from "@/lib/repositories/prisma-features.repository";
+import { UpdateManagedUserInput } from "@/modules/users/inputs/update-managed-user.input";
+import { UserWithProfile, UsersRepository } from "@/modules/users/users.repository";
+import { Injectable } from "@nestjs/common";
+
+import { sendChangeOfEmailVerification } from "@calcom/platform-libraries/emails";
+import type { Prisma } from "@calcom/prisma/client";
+
+export interface UpdateMeResult {
+  updatedUser: UserWithProfile;
+}
+
+@Injectable()
+export class MeService {
+  constructor(
+    private readonly usersRepository: UsersRepository,
+    private readonly schedulesService: SchedulesService_2024_04_15,
+    private readonly featuresRepository: PrismaFeaturesRepository
+  ) {}
+
+  async updateMe(params: {
+    user: UserWithProfile;
+    updateData: UpdateManagedUserInput;
+  }): Promise<UpdateMeResult> {
+    const { user, updateData } = params;
+    const update = { ...updateData };
+
+    const emailVerification = await this.featuresRepository.checkIfFeatureIsEnabledGlobally(
+      "email-verification"
+    );
+
+    const hasEmailBeenChanged = update.email && user.email !== update.email;
+    const newEmail = update.email;
+    let sendEmailVerification = false;
+
+    if (hasEmailBeenChanged && newEmail && !user.isPlatformManaged && emailVerification) {
+      const secondaryEmail = await this.usersRepository.findVerifiedSecondaryEmail(user.id, newEmail);
+
+      if (!secondaryEmail?.emailVerified) {
+        update.metadata = {
+          ...(user.metadata as Prisma.JsonObject),
+          ...(update.metadata || {}),
+          emailChangeWaitingForVerification: newEmail.toLowerCase(),
+        };
+
+        delete update.email;
+        sendEmailVerification = true;
+      }
+    }
+
+    const updatedUser = await this.usersRepository.update(user.id, update);
```

Comment:
If we have unverified email that we delete `delete update.email`, then there is no need to run `this.usersRepository.update` - we should only run if the update data actually contains something.